perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_facility_attribute_def_def_m365AllowedLicensesPriorities.java

```diff
@@ -0,0 +1,100 @@
+import cz.metacentrum.perun.core.implApi.modules.attributes.FacilityAttributesModuleImplApi;
+
+import java.util.LinkedHashMap;
+import java.util.List;
+
+/**
+ * Checks that value of m365AllowedLicensesPriorities is valid:
+ *   -	does not contain a null key
+ *   -	when a change is made, there is no m365LicenseGroup attribute under the same facility
+ *   	containing a license that no longer exists in this map
+ *
+ * @author Michal Berky <michal.berky@cesnet.cz>
+ */
+public class urn_perun_facility_attribute_def_def_m365AllowedLicensesPriorities extends FacilityAttributesModuleAbstract implements FacilityAttributesModuleImplApi {
+
+	@Override
+	public void checkAttributeSyntax(PerunSessionImpl perunSession, Facility facility, Attribute attribute) throws WrongAttributeValueException {
+		LinkedHashMap<String, String> licensesPriorities = attribute.valueAsMap();
+		if(licensesPriorities == null) return;
+
+		for (String key : licensesPriorities.keySet()) {
+			if (key == null) {
+				throw new WrongAttributeValueException("There can't be a null key in: " + attribute);
+			}
+		}
+	}
+
+	@Override
+	public void checkAttributeSemantics(PerunSessionImpl perunSession, Facility facility, Attribute attribute) throws WrongReferenceAttributeValueException, WrongAttributeAssignmentException {
+		LinkedHashMap<String, String> licensesMap = attribute.valueAsMap();
+		List<Resource> resources = fetchResourcesForFacility(perunSession, facility);
+
+		if (licensesMap == null) {
+			handleNullLicensesMap(perunSession, resources, attribute);
+			return;
+		}
+
+		verifyResourcesLicense(perunSession, resources, licensesMap);
+	}
+
+	private List<Resource> fetchResourcesForFacility(PerunSessionImpl perunSession, Facility facility) throws WrongReferenceAttributeValueException {
+		try {
+			return perunSession.getPerunBl().getFacilitiesManagerBl().getAssignedResources(perunSession, facility);
+		} catch (Exception e) {
+			throw new WrongReferenceAttributeValueException("Couldn't fetch resources from facility " + facility.getName(), e);
+		}
+	}
+
+	private void handleNullLicensesMap(PerunSessionImpl perunSession, List<Resource> resources, Attribute attribute) throws WrongReferenceAttributeValueException {
+		if (resources.isEmpty()) return;
+
+		for (Resource res : resources) {
+			Attribute resourceLicense = fetchResourceLicense(perunSession, res);
+			if (resourceLicense.getValue() != null) {
+				throw new WrongReferenceAttributeValueException("The attribute '" + attribute +
+					"' is being cleared, but there is still a resource in this facility depending on it: '" + res.getName());
+			}
+		}
+	}
+
+	private Attribute fetchResourceLicense(PerunSessionImpl perunSession, Resource res) throws WrongReferenceAttributeValueException {
+		try {
+			return perunSession.getPerunBl().getAttributesManagerBl().getAttribute(
+				perunSession, res, AttributesManager.NS_RESOURCE_ATTR_DEF + ":m365LicenseGroup");
```

Comment:
We usually extract dependant attribute names to a constant field.
Use it then to implement the `getDependencies()` method, which is missing.